src/prpy/base/robot.py

```diff
@@ -438,18 +437,33 @@ def ExecuteTrajectory(self, traj, defer=False, timeout=None, period=0.01, **kwar
         @param period poll rate, in seconds, for checking trajectory status
         @return trajectory executed on the robot
         """
+        # Don't execute trajectories that don't have at least one waypoint.
+        if traj.GetNumWaypoints() <= 0:
+            raise ValueError('Trajectory must contain at least one waypoint.')
+
+        # Check that the current configuration of the robot matches the
+        # initial configuration specified by the trajectory.
+        if not util.IsAtTrajectoryStart(self, traj):
+            raise exceptions.TrajectoryAborted(
+                'Trajectory started from different configuration than robot.')
+
+        # If there was only one waypoint, at this point we are done!
+        if traj.GetNumWaypoints() == 1:
+            return traj
```

Comment:
This will break if `defer=True`. In that case, you should return a `Future` that yields `traj`, instead of returning `traj` itself.